src/main/java/com/google/summit/ast/TypeRef.kt

```diff
@@ -69,6 +69,18 @@ class TypeRef(val components: List<Component>, val arrayNesting: Int) :
       components.joinToString(".") { it.asCodeString() } + "[]".repeat(arrayNesting)
     }
 
+  /**
+   * Returns the type reference as a type-erased string.
+   *
+   * This is the same as [asCodeString] but without any type arguments.
+   */
+  fun asTypeErasedString(): String =
+    if (isVoid()) {
+      "void"
```

Comment:
I see this same behavior with the method above as well, and I'm a bit confused why "void", and no other type, needs to be treated specially.

Reply:
I don't know that it needs to be. There are other internal representation choices that could work just as well. It is treated differently because:
1. In the parser, it is the only type reference that is a language keyword and not an identifier. The grammar rules therefore handle it specially.
2. In static analysis, there are quite a few situations where void is different than an arbitrary return type. (E.g. "Should this return statement have a value or not?")